src/partition_mesh.py

```diff
@@ -83,387 +45,423 @@ def __str__(self):
+                directory = os.path.abspath(self.args.directory)
+                os.makedirs(directory, exist_ok=True)
+                out_meshname = os.path.join(directory, self.args.out_meshname)
             else:
-                n //= i
-                factors.append(i)
-        if n > 1:
-            factors.append(n)
-        return factors
-
-    def greedy_choose(factors):
-        """ Greedily choose "best" divisors"""
-        small = big = 1
-        for factor in reversed(factors):
-            if big <= small:
-                big *= factor
+                out_meshname = self.args.out_meshname
+            extension = os.path.splitext(mesh_name)[1]
+            if extension == ".vtk":
+                shutil.copy(mesh_name, out_meshname + ".vtk")
+                return
+            elif extension == ".vtu":
+                self.vtu2vtk(mesh_name, out_meshname)
+                return
             else:
-                small *= factor
-        return small, big
-
-    small, big = greedy_choose(prime_factors(numparts))
-    small_interval = (max_point[small_dim] - min_point[small_dim]) / small
-    big_interval = (max_point[big_dim] - min_point[big_dim]) / big
-    labels = []
-    logging.info("Uniform partioning of mesh size {} into {} x {} partitions.".format(
-        len(mesh), small, big))
-    for point in mesh:
-        small_offset = point[small_dim] - min_point[small_dim]
-        small_index = int(small_offset / small_interval)
-        small_index = min(small_index, small - 1)
-        big_offset = point[big_dim] - min_point[big_dim]
-        big_index = int(big_offset / big_interval)
-        big_index = min(big_index, big - 1)
-        partition_num = small_index * big + big_index
-        labels.append(partition_num)
-    return labels
-
-
-def apply_partition(orig_mesh: Mesh, part, numparts: int):
-    """
-    Partitions a mesh into many meshes when given a partition and a mesh.
-    """
-    meshes = [Mesh() for _ in range(numparts)]
-    mapping = {}  # Maps global index to partition and local index
-    for i, point in enumerate(orig_mesh.points):
-        partition = part[i]
-        selected = meshes[partition]
-        mapping[i] = (partition, len(selected.points))
-        selected.points.append(point)
-        selected.data_index.append(i)
-
-    assert(len(mapping) == len(orig_mesh.points))
-    assert(len(orig_mesh.cells) == len(orig_mesh.cell_types))
-    # Save discarded cells and their types to allow recovery
-    discardedCells = []
-    discardedCellTypes = []
-    for cell, type in zip(orig_mesh.cells, orig_mesh.cell_types):
-        partitions = list(map(lambda idx: mapping[idx][0], cell))
-        if len(set(partitions)) == 1:
-            meshes[partitions[0]].cells.append(tuple([mapping[gidx][1] for gidx in cell]))
-            meshes[partitions[0]].cell_types.append(type)
+                raise Exception(f"Unkown input file extension {extension} please check your input file.")
```

Comment:
Is the "f" here a typo or meaningful?

Reply:
> Looks good overall.
> 
> Two main points:
> 
> 1. You use some type of formatting for the python scripts. I guess we should enforce some standard here and put it into a GH workflow. I think @BenjaminRodenberg has already some experience with this.

Currently using autopep8 for formatting. Let's add to GH workflow.

> 2. How do you test these changes? I know that we have some basic tests, but did you also run the turbine case or something similar?

I tested. Working fine with the turbine.

Reply:
f-string in python for formatting, {extension} is a variable.